modules/RTC/TraceablePeerConnection.js

```diff
@@ -1596,17 +1599,17 @@ TraceablePeerConnection.prototype._assertTrackBelongs = function(
 };
 
 /**
- * Returns the codec that is configured on the client as the preferred video codec.
- * This takes into account the current order of codecs in the local description sdp.
+ * Returns the codec that is configured on the client as the preferred video codec for the given local video track.
+ *
+ * @param {JitsiLocalTrack} localTrack - The local video track.
+ * @returns {CodecMimeType} The codec that is set as the preferred codec for the given local video track.
  *
- * @returns {CodecMimeType} The codec that is set as the preferred codec to receive
- * video in the local SDP.
  */
-TraceablePeerConnection.prototype.getConfiguredVideoCodec = function() {
-    const localVideoTrack = this.getLocalVideoTracks()[0];
+TraceablePeerConnection.prototype.getConfiguredVideoCodec = function(localTrack) {
+    const localVideoTrack = localTrack ?? this.getLocalVideoTracks()[0];
+    const rtpSender = this.findSenderForTrack(localVideoTrack.getTrack());
 
-    if (this.usesCodecSelectionAPI() && localVideoTrack) {
-        const rtpSender = this.findSenderForTrack(localVideoTrack.getTrack());
+    if (this.usesCodecSelectionAPI() && rtpSender) {
```

Comment:
We need the guardrail around the sender even though `RTCPeerConnection::getSenders()` is supported on all our endpoints because we remove the track from the sender when we mute the video source. So `findSenderForTrack` will return `undefined` in this case.